e2e/cypress/integration/api/iam/12_projects_api_node_ingestion.spec.ts

```diff
@@ -217,67 +217,92 @@ describeIfIAMV2p1('Nodemanager ingestion project tagging', () => {
     });
 
     // wait for the report to be ingested
-    waitForNodemanagerNode(complianceNodeId, 30);
+    cy.waitForNodemanagerNode(complianceNodeId, 30);
+    cy.waitForComplianceNode(complianceNodeId, start, end, 30);
     // wait for the client run report to be ingested
-    waitForNodemanagerNode(clientRunsNodeId, 30);
+    cy.waitForNodemanagerNode(clientRunsNodeId, 30);
+    cy.waitForClientRunsNode(clientRunsNodeId, 30);
   });
 
   after(() => cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']));
 
-  // Ensure that both the compliance and client run nodes' tags are updated
-  for ( const projectWithRule of projectsWithRule ) {
-    it(`when a project has a rule that matches a node's ${projectWithRule.rule.name},
-    successfully associates that node with the project`, () => {
-      cy.request({
-        headers: {
-          projects: [projectWithRule.project.id],
-          'api-token': Cypress.env('ADMIN_TOKEN')
-        },
-        method: 'POST',
-        url: '/api/v0/nodes/search',
-        body: {
-          order: 'DESC',
-          page: 1,
-          per_page: 100
-        }
-      }).then((response) => {
-        expect(response.body.nodes.length).to.be.greaterThan(0);
-        expect(response.body.nodes.some((node: any) =>
-          node.id === complianceNodeId)).to.equal(true);
-        expect(response.body.nodes.some((node: any) =>
-          node.id === clientRunsNodeId)).to.equal(true);
+  describe('Node Manager', () => {
+    // Ensure that both the compliance and client run nodes' tags are updated
```

Comment:
maybe add to this: `by fetching the nodes with the expected project filter` (and repeat for the similar comments in the rest of the file)